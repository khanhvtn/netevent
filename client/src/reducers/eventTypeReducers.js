import {
  EVENT_TYPE_GET_ALL,
  EVENT_TYPE_LOADING,
  EVENT_TYPE_CREATE_SUCCESS,
  EVENT_TYPE_UPDATE_SUCCESS,
  EVENT_TYPE_DELETE_SUCCESS,
  EVENT_TYPE_GET_ALL_FILTER,
  EVENT_TYPE_RECOVERY_SUCCESS,
} from '../constants';

const initialState = {
  isLoading: false,
  eventTypes: [],
  totalPages: null,
  createSuccess: false,
  updateSuccess: false,
  deleteSuccess: false,
  recoverySuccess: false,
};

export default function facilityReducers(state = initialState, action) {
  switch (action.type) {
    case EVENT_TYPE_GET_ALL:
      return {
        ...state,
        eventTypes: action.payload.data.data,
      };
    case EVENT_TYPE_GET_ALL_FILTER:
      return {
        ...state,
        eventTypes: action.payload.data.data,
        totalPages: action.payload.data.totalPages,
      };
    case EVENT_TYPE_LOADING:
      return { ...state, isLoading: action.payload };
    case EVENT_TYPE_CREATE_SUCCESS:
      return { ...state, createSuccess: action.payload };
    case EVENT_TYPE_UPDATE_SUCCESS:
      return { ...state, updateSuccess: action.payload };
    case EVENT_TYPE_DELETE_SUCCESS:
      return { ...state, deleteSuccess: action.payload };
    case EVENT_TYPE_RECOVERY_SUCCESS:
      return { ...state, recoverySuccess: action.payload };
    default:
      return state;
  }
}
