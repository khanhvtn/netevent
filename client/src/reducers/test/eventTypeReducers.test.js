import {
    EVENT_TYPE_GET_ALL,
    EVENT_TYPE_LOADING,
    EVENT_TYPE_CREATE_SUCCESS,
    EVENT_TYPE_UPDATE_SUCCESS,
    EVENT_TYPE_DELETE_SUCCESS,
    EVENT_TYPE_GET_ALL_FILTER,
} from '../../constants';
import eventTypeReducers from '../eventTypeReducers'
import expect from 'expect'

describe('EVENT TYPE REDUCER', () => {
    const initialState = {
        isLoading: false,
        eventTypes: [],
        totalPages: null,
        createSuccess: false,
        updateSuccess: false,
        deleteSuccess: false,
    };
    it('Should return default state', () => {
        const action = eventTypeReducers(undefined, {})
        expect(action).toEqual(initialState)
    })
    it('Should return EVENT_TYPE_LOADING', () => {
        const action = eventTypeReducers(initialState, {
            type: EVENT_TYPE_LOADING
            })
        expect(action).toEqual({
            isLoading: action.payload,
            eventTypes: [],
            totalPages: null,
            createSuccess: false,
            updateSuccess: false,
            deleteSuccess: false,
        })
    })
    it('Should return EVENT_TYPE_CREATE_SUCCESS', () => {
        const action = eventTypeReducers(initialState, {
            type: EVENT_TYPE_CREATE_SUCCESS
            })
        expect(action).toEqual({
            isLoading: false,
            eventTypes: [],
            totalPages: null,
            createSuccess: action.payload,
            updateSuccess: false,
            deleteSuccess: false,
        })
    })
    it('Should return EVENT_TYPE_UPDATE_SUCCESS', () => {
        const action = eventTypeReducers(initialState, {
            type: EVENT_TYPE_UPDATE_SUCCESS
            })
        expect(action).toEqual({
            isLoading: false,
            eventTypes: [],
            totalPages: null,
            createSuccess: false,
            updateSuccess: action.payload,
            deleteSuccess: false,
        })
    })
    it('Should return EVENT_TYPE_DELETE_SUCCESS', () => {
        const action = eventTypeReducers(initialState, {
            type: EVENT_TYPE_DELETE_SUCCESS
            })
        expect(action).toEqual({
            isLoading: false,
            eventTypes: [],
            totalPages: null,
            createSuccess: false,
            updateSuccess: false,
            deleteSuccess: action.payload,
        })
    })
})