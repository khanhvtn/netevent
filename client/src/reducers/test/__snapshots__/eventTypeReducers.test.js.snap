// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EVENT TYPE REDUCER Should return EVENT_TYPE_CREATE_SUCCESS 1`] = `
Object {
  "createSuccess": undefined,
  "deleteSuccess": false,
  "eventTypes": Array [],
  "isLoading": false,
  "totalPages": null,
  "updateSuccess": false,
}
`;

exports[`EVENT TYPE REDUCER Should return EVENT_TYPE_DELETE_SUCCESS 1`] = `
Object {
  "createSuccess": false,
  "deleteSuccess": undefined,
  "eventTypes": Array [],
  "isLoading": false,
  "totalPages": null,
  "updateSuccess": false,
}
`;

exports[`EVENT TYPE REDUCER Should return EVENT_TYPE_LOADING 1`] = `
Object {
  "createSuccess": false,
  "deleteSuccess": false,
  "eventTypes": Array [],
  "isLoading": undefined,
  "totalPages": null,
  "updateSuccess": false,
}
`;

exports[`EVENT TYPE REDUCER Should return EVENT_TYPE_UPDATE_SUCCESS 1`] = `
Object {
  "createSuccess": false,
  "deleteSuccess": false,
  "eventTypes": Array [],
  "isLoading": false,
  "totalPages": null,
  "updateSuccess": undefined,
}
`;

exports[`EVENT TYPE REDUCER Should return default state 1`] = `
Object {
  "createSuccess": false,
  "deleteSuccess": false,
  "eventTypes": Array [],
  "isLoading": false,
  "totalPages": null,
  "updateSuccess": false,
}
`;
